### PR TITLE
Make it easy to turn off xwayland

### PR DESCRIPTION
**Pull request by RAVENz46**
Like hyprland, sway and river, this make it easy to toggle xwayland on/off.

### CODE REVIEW

**Comment by RAVENz46** [comment]
Should I use `config.services.desktopManager.cosmic.xwayland.enable or true` instead?
I realized that IDK what happen when install cosmic-comp as home-manager packages.

**Comment by RAVENz46** [comment]
I believe this method better.

File: nixos/cosmic/module.nix [hunk 1]
```diff
@@ -9,6 +9,8 @@ in
   options = {
     services.desktopManager.cosmic = {
       enable = lib.mkEnableOption (lib.mdDoc "COSMIC desktop environment");
+
+      xwayland.enable = lib.mkEnableOption "XWayland" // { default = true; };
```
Comment (lilyinstarlight) [inline nixos/cosmic/module.nix:13]:
```suggestion
      xwayland.enable = lib.mkEnableOption "Xwayland support for cosmic-comp" // { default = true; };
```

**Review by lilyinstarlight — COMMENTED** [review]
One minor nit and this otherwise looks good, thank you!

File: nixos/cosmic/module.nix (hunk 1 above)
Comment (RAVENz46) [reply nixos/cosmic/module.nix]: Done:)

**Comment by bbigras** [comment]
Any progress on this?

**Comment by RAVENz46** [comment]
@bbigras Ready to merge.
Just wait.
Temporary, you can disable like this.
```nix
  environment = {
    systemPackages = with pkgs; [
      (cosmic-comp.override {
        useXWayland = false;
      })
    ];
    cosmic.excludePackages = with pkgs; [
      cosmic-comp
    ];
  };
```

**Review by lilyinstarlight — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**